File: library/src/main/java/com/healthy/library/contract/LotteryContract.java
package com.healthy.library.contract;

import com.healthy.library.base.BasePresenter;
import com.healthy.library.base.BaseView;
import com.healthy.library.model.LotteryModel;
import com.healthy.library.model.UpdatePatchVersion;

import java.util.Map;

/**
 * @author Li
 * @date 2019/03/27 11:20
 * @des
 */

public interface LotteryContract {
    interface View extends BaseView {
        void onLotteryInfoSuccess(LotteryModel lotteryModel);

    }


    interface Presenter extends BasePresenter {
        void getLotteryInfo(Map<String,Object> map);
    }
}

File: hmm-servicecenter/src/main/java/com/health/servicecenter/contract/LotteryContract.kt
package com.health.servicecenter.contract

import com.healthy.library.base.BasePresenter
import com.healthy.library.base.BaseView
import com.healthy.library.model.LotteryInfoModel

interface LotteryContract {

    interface View : BaseView {

        fun onLotteryInfoSuccess(lotteryInfoModel: LotteryInfoModel?)

        fun onCheckLotterySuccess(triggerPrivilege: Int)

    }

    interface Presenter : BasePresenter {

        /**
         * 获取当前活动触发条件详情
         */
        fun getLotteryInfo(map: MutableMap<String, Any>)

        /**
         * 检测是否获取签到抽奖的资格
         */
        fun checkLottery(map: MutableMap<String, Any>)

    }
}